endpoints/event.go

```diff
@@ -0,0 +1,367 @@
+package endpoints
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/julienschmidt/httprouter"
+	"github.com/prebid/prebid-server/analytics"
+	"github.com/prebid/prebid-server/cache"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type Type string
+type Format string
+type Analytics string
+
+const (
+	WIN Type = "win"
+	IMP Type = "imp"
+
+	BLANK Format = "b"
+	IMAGE Format = "i"
+
+	ENABLED  Analytics = "1"
+	DISABLED Analytics = "0"
+
+	// Required
+	TEMPLATE_URL         = "%v/event?t=%v&b=%v&a=%v"
+	TYPE_PARAMETER       = "t"
+	BID_ID_PARAMETER     = "b"
+	ACCOUNT_ID_PARAMETER = "a"
+
+	// Optional
+	BIDDER_PARAMETER    = "bidder"
+	TIMESTAMP_PARAMETER = "ts"
+	FORMAT_PARAMETER    = "f"
+	ANALYTICS_PARAMETER = "x"
+
+	TRACKING_PIXEL_PNG              = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAABHNCSVQICAgIfAhkiAAAAA1JREFUCJljYGBgYAAAAAUAAYehTtQAAAAASUVORK5CYII="
+	TRACKING_PIXEL_PNG_CONTENT_TYPE = "image/png"
+)
+
+type EventRequest struct {
+	Type      Type      `json:"type,omitempty"`
+	Bidid     string    `json:"bidid,omitempty"`
+	AccountId string    `json:"account_id,omitempty"`
+	Bidder    string    `json:"bidder,omitempty"`
+	Timestamp int64     `json:"timestamp,omitempty"`
+	Format    Format    `json:"format,omitempty"`
+	Analytics Analytics `json:"analytics,omitempty"`
+}
+
+type TrackingPixel struct {
+	Content     []byte `json:"content,omitempty"`
+	ContentType string `json:"content_type,omitempty"`
+}
+
+type eventEndpoint struct {
+	DataCache     cache.Cache
+	Analytics     analytics.PBSAnalyticsModule
+	TrackingPixel *TrackingPixel
+}
+
+func NewEventEndpoint(dataCache cache.Cache, analytics analytics.PBSAnalyticsModule) httprouter.Handle {
+	ee := &eventEndpoint{
+		DataCache:     dataCache,
+		Analytics:     analytics,
+		TrackingPixel: loadTrackingPixel(),
+	}
+
+	return ee.Handle
+}
+
+func (e *eventEndpoint) Handle(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	// parse event request from http req
+	eventRequest, err := ParseEventRequest(r)
+
+	// handle possible parsing errors
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("invalid request: %s\n", err.Error())))
+
+		return
+	}
+
+	// validate account id
+	accountId, err := validateRequiredParameter(r, ACCOUNT_ID_PARAMETER)
+
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte(fmt.Sprintf("Account '%s' is required query parameter and can't be empty", ACCOUNT_ID_PARAMETER)))
+		return
+	}
+	eventRequest.AccountId = accountId
+
+	if eventRequest.Analytics == ENABLED {
```

Comment:
Style: There's a lot of code intended below this if check, please consider a reverted check with a quick return:

```
if eventRequest.Analytics != ENABLED {
  w.WriteHeader(http.StatusUnauthorized) // not sure what this should return
  return
}
```